src/routes/shared/reports/delete.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @memberof module:router~mainRouter~reportsRouter
+ * @inner
+ * @namespace reports
+ */
+
+const { default: mongoose } = require('mongoose')
+const { Reports } = require('../../../models/reports')
+
+/**
+ * Main reports function
+ * @name DELETE /shared/reports
+ * @function
+ * @memberof module:router~mainRouter~reportsRouter~reports
+ * @inner
+ * @async
+ * @param {Object} req
+ * @param {Object} res
+ * @returns 200 if OK and return access token and role name
+ * @returns 400 if Invalid arguments
+ * @returns 500 if Internal Server Error
+ */
+module.exports = async (req, res) => {
+  try {
+    // Verif received data
+    const id = req.params.id
+
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+      return res.status(400).json({ message: 'Invalid request' })
+    }
+
+    const report = await Reports.findById(id)
+
+    if (!report) {
+      return res.status(400).json({ message: 'Invalid request' })
+    }
+
+    await Reports.findByIdAndDelete(id)
```

Comment:
Merge the two queries like this: 
`
if (!await Reports.findByIdAndDelete(id)) {
      return res.status(400).json({ message: 'Invalid request' })
}
`